packages/wallet-core/src/serde/wire-format/deserialize.ts

```diff
@@ -29,13 +30,20 @@ export function convertToInternalParticipant(participant: {
   return {...participant, destination: makeDestination(participant.destination)};
 }
 
+export function validatePayload(rawPayload: unknown): Payload {
+  // todo: wire-format should export a validator specially for the payload
+  return deserializeMessage(validateMessage({recipient: '', sender: '', data: rawPayload}));
```

Comment:
This call to `deserializeMessage` calls `deserializeState`, which calls `getAddress` on the signatures.

Reply:
We have to do this later anyway in the wallet, when we check the signature.